src/FSharp.Core/printf.fs

```diff
@@ -658,8 +658,28 @@ module internal PrintfImpl =
     
     /// Contains functions to handle left/right and no justification case for numbers
     module GenericNumber =
-
-        let isPositive (n: obj) = 
+        
+        let inline doubleIsPositive (n: double) =
+            n >= 0.0
+                // Ensure -0.0 is treated as negative (see https://github.com/dotnet/fsharp/issues/15557)
+                // and use bitwise comparison because floating point comparison treats +0.0 as equal to -0.0
+                && (BitConverter.DoubleToInt64Bits n <> BitConverter.DoubleToInt64Bits -0.0)
```

Comment:
Is this going to do the right thing for `NaN`? What is the behavior F# expects to have there, particularly with the existance of both `+NaN` and `-NaN` at the bitwise level?